shtdlib.sh

```diff
@@ -2526,26 +2533,47 @@ function associate_array {
     done
 }
 
-# Safely loads config file
+# Safely loads config file or config from stdin
 # First parameter is filename, all consequent parameters are assumed to be
 # valid configuration parameters
+# By default settings are printed when config is piped since variables set
+# would only live in a subshell, this can be changed using the "print_settings"
+# variable
 function load_config {
-    config_file="${1}"
-    # Verify config file permissions are correct and warn if they aren't
-    # Dual stat commands to work with both linux and bsd
-    while read -r line; do
+    # If first argument is a filename read it and pass it to the function
+    if [ -f "${1:-}" ]; then
+        print_settings='false'
```

Comment:
Should we allow flexibility here to allow printing? 

E.g.
```suggestion
        print_settings="${print_settings:-false}"
```

So we can do something like:
```
print_settings=true load_config  config foo
Loaded config parameter foo with value of 'bar'
foo=bar
```

Reply:
We do .. and we don't.

Print Setting will actually cause issues because we're allowing piping but it does not matter because that particular case will recurse so it will pick up the setting on the second run :)